bot_modules/irc_kudos.js

```diff
@@ -4,13 +4,17 @@ var what_max_length = 32;
 
 var process = function(words, operator) {
 	// build what
-	what = words.join(' ');
+	var what = words.join(' ');
 	// shave operator off
-	what = what.substr(0, what.length - 2).trim();
+	what = what.substr(0, what.length - 2);
```

Comment:
is `what.length <= 2` accounted for here ?

Reply:
i think it just silently throws an exception and doesn't execute, or it might not get called.

Reply:
iirc these are meant to remove extra text added by the irc protocol
i could be wrong and putting a check in isn't a bad idea

Reply:
do you know what the two characters normally are?
it could check for them, and remove them if they exist

Reply:
uh... i think they were two spaces but i don't clearly rememeber -- have to remove the substr command to see

Reply:
just to be clear this doesn't crash or anything
![screen shot 2018-03-21 at 3 11 33 pm](https://user-images.githubusercontent.com/6981186/37732041-3036fbb2-2d1a-11e8-8ecf-ea517002e8e2.png)

Reply:
that's cool